### ISSUE TITLE
S2303 FP in exception message

### ISSUE THREAD

**Issue by lg2de** [SonarSource/sonar-dotnet#3246]
### Description

The implementation for rule S2302 is too aggressive. It criticizes words of natural language.

Wouldn't it be better to apply the check only to the parameter "paramName"?

### Repro steps

```cs
if (message == null)
  throw new ArgumentNullException(nameof(message), "The message should be correct.");
```

### Expected behavior

No issue.

### Actual behavior

`Replace the string 'message' with 'nameof(message)'.`

### Known workarounds

```cs
if (message == null)
  throw new ArgumentNullException(nameof(message), $"The {nameof(message)} should be correct.");
```

### Related information

* SonarC# Version: current master

**Comment by pavel-mikula-sonarsource** [comment]
Hello @lg2de ,

This is a clear False-Positive in your case, but it cannot be fixed unfortunately. There's no way to distinguish natural language word from variable name when they're accidentally the same. Purpose of this rule is to detect variable name in the message argument as well to avoid inconsistency in the code.  Consider this case:
```C#
if (message == null)
  throw new ArgumentNullException(nameof(message), 
    "The message argument should be a real message.");
```
It should be replaced by this
```C#
if (message == null)
  throw new ArgumentNullException(nameof(message), 
    $"The {nameof(message)} argument should be a real message.");
```
but that would still raise the rule on the last word.
